cdap-gateway/src/main/java/io/cdap/cdap/gateway/router/NettyRouter.java

```diff
@@ -126,7 +126,8 @@ public NettyRouter(CConfiguration cConf, SConfiguration sConf, @Named(Constants.
 
   @Override
   protected void startUp() throws Exception {
-    if (SecurityUtil.isManagedSecurity(cConf)) {
+    // If internal authorization enforcement is enabled, we avoid re-initialization of the token manager.
+    if (SecurityUtil.isManagedSecurity(cConf) && !cConf.getBoolean(Constants.Security.ENFORCE_INTERNAL_AUTH)) {
```

Comment:
Yeah agreed. I think the best way to do this is to refactor the TokenValidator and then ensure that the TokenManager lifecycle is managed at the service main level. However, we might want to defer that refactor to a later PR since this one is already sizable as it is. I've gone ahead and applied your suggestion here.

Reply:
Actually I don't think it's feasible to do what you've described, so I went ahead and performed the refactor. There are too many edge cases in which the TokenManager may (or may not) be needed which, right now, must be determined at the service main level.

Reply:
Addressed in part by the refactor in #13522. Will continue to improve down the line.